isis/src/base/objs/Cube/Cube.cpp

```diff
@@ -829,6 +830,18 @@ namespace Isis {
     m_ioHandler->read(bufferToFill);
   }
 
+  History *Cube::readHistory(const QString &name) const {
+    Blob historyBlob(name, "History");
+    try {
+      // read history from cube, if it exists.
+      historyBlob.Read(fileName());
+    }
+    catch (IException &) {
+    // if the history does not exist in the cube, this function creates it.
+    }
+    History *history = new History(historyBlob);
```

Comment:
Is there a reason why this is returning a pointer? I'm concerned about memory leaks here.

Reply:
I will update this to not return a pointer